src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/resources/RuntimeHelper.java.template

```diff
@@ -34,7 +34,7 @@ final class RuntimeHelper {
         return obj;
     }
 
-    static SegmentAllocator DEFAULT_ALLOCATOR = MemorySegment::allocateNative;
+    private final static SegmentAllocator DEFAULT_ALLOCATOR = (x, y) -> MemorySegment.allocateNative(x, y, ResourceScope.newImplicitScope());
```

Comment:
If this is only used in the varargs case, then I think having a THROWING_ALLOCATOR which always throws is also an option as I don't think we really want this to ever be used.